src/Aspire.Hosting/Dashboard/DcpDataSource.cs

```diff
@@ -104,6 +111,146 @@ await foreach (var (eventType, resource) in _kubernetesService.WatchAsync<T>(can
+                CreationTimeStamp = DateTime.UtcNow,
+                Image = containerImageAnnotation.Image,
+                State = "Starting",
+                ContainerId = null,
+                ExitCode = null,
+                ExpectedEndpointsCount = null,
+                Environment = [],
+                Endpoints = [],
+                Services = [],
+                Command = null,
+                Args = [],
+                Ports = []
+            };
+
+            _placeHolderResources.TryAdd(resource.Name, snapshot);
+
+            await _onResourceChanged(snapshot, ResourceSnapshotChangeType.Upsert).ConfigureAwait(false);
+        }
+        else if (resource is ProjectResource p)
+        {
+            var snapshot = new ProjectSnapshot
+            {
+                Name = p.Name,
+                DisplayName = p.Name,
+                Uid = p.Name,
+                CreationTimeStamp = DateTime.UtcNow,
+                ProjectPath = p.GetProjectMetadata().ProjectPath,
+                State = "Starting",
+                ExpectedEndpointsCount = null,
+                Environment = [],
+                Endpoints = [],
+                Services = [],
+                ExecutablePath = null,
+                ExitCode = null,
+                Arguments = null,
+                ProcessId = null,
+                StdErrFile = null,
+                StdOutFile = null,
+                WorkingDirectory = null
+            };
+
+            _placeHolderResources.TryAdd(resource.Name, snapshot);
+
+            await _onResourceChanged(snapshot, ResourceSnapshotChangeType.Upsert).ConfigureAwait(false);
+        }
+        else if (resource is ExecutableResource exe)
+        {
+            var snapshot = new ExecutableSnapshot
+            {
+                Name = exe.Name,
+                DisplayName = exe.Name,
+                Uid = exe.Name,
+                CreationTimeStamp = DateTime.UtcNow,
+                ExecutablePath = exe.Command,
+                WorkingDirectory = exe.WorkingDirectory,
+                Arguments = null,
+                State = "Starting",
+                ExitCode = null,
+                StdOutFile = null,
+                StdErrFile = null,
+                ProcessId = null,
+                ExpectedEndpointsCount = null,
+                Environment = [],
+                Endpoints = [],
+                Services = []
+            };
+
+            _placeHolderResources.TryAdd(resource.Name, snapshot);
+
+            await _onResourceChanged(snapshot, ResourceSnapshotChangeType.Upsert).ConfigureAwait(false);
+        }
+        else if (resource.TryGetLastAnnotation<DashboardAnnotation>(out var dashboardAnnotation))
+        {
+            // We have a dashboard annotation, so we want to create a snapshot for the resource
+            // and update data immediately. We also want to watch for changes to the dashboard state.
+            var state = dashboardAnnotation.GetIntialState();
```

Comment:
Intial -> Initial

Is this method needed? I wonder if it could just be rolled up into WatchAsync. i.e. the initial state is returned immediately from WatchAsync.

Reply:
It's needed because the producer needs to make state changes based on the initial snapshot